services/src/main/java/org/fao/geonet/api/site/SiteInformation.java

```diff
@@ -226,6 +226,13 @@ private void loadDatabaseInfo(ServiceContext context) throws SQLException {
             connection = context.getBean(DataSource.class).getConnection();
             dbURL = connection.getMetaData().getURL();
             databaseProperties.put("db.openattempt", "Database Opened Successfully");
+            databaseProperties.put("db.type", connection.getMetaData().getDatabaseProductName());
+            databaseProperties.put("db.version", connection.getMetaData().getDatabaseProductVersion());
+            databaseProperties.put("db.driver", connection.getMetaData().getDriverName());
+            databaseProperties.put("db.driverVersion", connection.getMetaData().getDriverVersion());
+            databaseProperties.put("db.username", connection.getMetaData().getUserName());
+            databaseProperties.put("db.schema", connection.getSchema());
```

Comment:
When using a JNDI connection, this method throws an exception:

```
2023-10-09T09:19:07,877 ERROR [geonetwork] - Handler dispatch failed; nested exception is java.lang.AbstractMethodError: Receiver class org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper does not define or inherit an implementation of the resolved method 'abstract java.lang.String getSchema()' of interface java.sql.Connection.
org.springframework.web.util.NestedServletException: Handler dispatch failed; nested exception is java.lang.AbstractMethodError: Receiver class org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper does not define or inherit an implementation of the resolved method 'abstract java.lang.String getSchema()' of interface java.sql.Connection.
```

See details to test with Jetty JNDI: https://geonetwork-opensource.org/manuals/3.12.x/fr/install-guide/configuring-database.html#configuring-a-database-via-jndi

Reply:
I tried to apply your fix however I get the following error.  I’m not sure if you were able to build the code with the suggested changes?

`org.apache.commons.dbcp2.PoolingDataSource.PoolGuardConnectionWrapper has private access in org.apache.commons.dbcp2.PoolingDataSource`

Maybe I can just do a try catch block and if it fails then it will not include the schema. It could add a message indicating that it could not determine schema.  Not sure if that should be done to all other fields as well?

Reply:
I'll recheck, I don't remember to get private access issues.

Reply:
I get the same issue @ianwallen, I was debugging in IntelliJ and didn't get that inside the IDE, but it happens when running the application.

A fix for this, unless a better option, ignoring the property if using JNDI:

```
            if (connection instanceof BasicDataSource) {
                databaseProperties.put("db.schema", connection.getSchema());
            }
```

Reply:
I did not think BasicDataSource supported getSchema()

Did you mean 

```
if (connection instanceof BasicDataSource) {
      databaseProperties.put("db.schema", "" + basicDataSource.getDefaultSchema());
} else {
     databaseProperties.put("db.schema", connection.getSchema());
}
```


I also found the sample code which suggest that the following may work?

```
if (connection instanceof DelegatingConnection) {
    databaseProperties.put("db.schema", ((DelegatingConnection)connection).getInnermostDelegate().getSchema());
} else {
    databaseProperties.put("db.schema", connection.getSchema());
}
```


I tried using the following sample and I get a regular connection and there are no issues.
[https://github.com/geonetwork/core-geonetwork/blob/main/web/src/main/webResources/WEB-INF/config-db/jndiExamples_tomcat/h2_JNDI_Example.xml](https://can01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgeonetwork%2Fcore-geonetwork%2Fblob%2Fmain%2Fweb%2Fsrc%2Fmain%2FwebResources%2FWEB-INF%2Fconfig-db%2FjndiExamples_tomcat%2Fh2_JNDI_Example.xml&data=05%7C01%7CIan.Allen%40dfo-mpo.gc.ca%7C7373250981794efc3da408dbd3c07677%7C1594fdaea1d94405915d011467234338%7C0%7C0%7C638336595810759640%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=2l7jF4zp6%2BMa3ZFKUXwuyO3is4YHQkVeY8pmRDiY4RM%3D&reserved=0)

https://github.com/geonetwork/core-geonetwork/blob/b417ead0bf936d3b43d16e99171c370ec96c84a3/web/src/main/webResources/WEB-INF/config-db/jndiExamples_tomcat/h2_JNDI_Example.xml#L26-L35
 
I’m trying to reproduce your issue to get `PoolingDataSource.PoolGuardConnectionWrapper`.  I used Tomcat 9.0.75 and don't see that class?

Reply:
@ianwallen I'm not sure if the following code works for you or not?

I tried and I got a `ClassCastException`, but I can recheck again.

```
if (connection instanceof DelegatingConnection) {
    databaseProperties.put("db.schema", ((DelegatingConnection)connection).getInnermostDelegate().getSchema());
} else {
    databaseProperties.put("db.schema", connection.getSchema());
}
```

My suggestion was to define the schema property when not using JNDI:

```
if (connection instanceof BasicDataSource) {
    databaseProperties.put("db.schema", connection.getSchema());
}
```

Reply:
@josegar74 
I was finally able to duplicate your problem using the jetty jndi with h2.

It seems like the issue is related to mismatching drivers.
    https://stackoverflow.com/questions/17969365/why-i-am-getting-java-lang-abstractmethoderror-errors

So I added a try catch block around the new properties to get the AbstractMethodError that could occur in this case.  This way any if any of the other properties has similar issues then it will simply not add the properties.

If you retest you will see all the other fields populated. The schema will not be populated until the driver issue is fixed.  

Not sure if the driver issue is related to an older version of jetty?